lib/blocs/unread_messages_cubit/unread_messages_cubit.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:math';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:twake/blocs/channels_cubit/channels_cubit.dart';
+import 'package:twake/blocs/messages_cubit/messages_cubit.dart';
+import 'package:twake/blocs/unread_messages_cubit/unread_messages_state.dart';
+import 'package:twake/models/message/message.dart';
+
+abstract class BaseUnreadMessagesCubit extends Cubit<UnreadMessagesState> {
+  late final BaseChannelsCubit _channelsCubit;
+
+  BaseUnreadMessagesCubit({BaseChannelsCubit? channelsCubit})
+      : super(UnreadMessagesInitial()) {
+    if (channelsCubit == null) {
+      channelsCubit = ChannelsCubit();
+    }
+    _channelsCubit = channelsCubit;
+  }
+
+  void listenOwnerMessage() {
```

Comment:
What? Why listenOwnerMessage emit NoUnreadMessages()?

Reply:
when user type his message, the UnreadMessage State is set to NoUnreadMessage( the state that don't have any unread messages)